lib/ActivityStreams.js

```diff
@@ -152,9 +152,11 @@ ActivityStreams.prototype = {
    * Process the passed in links, save them, get from cache and response to content.
    */
   _processAndSendLinks(links, responseType, append, worker, previewsOnly = false) {
+    const event = this._tabTracker.generateEvent({source: responseType});
     let processedLinks = this._previewProvider.processLinks(links);
-    this._previewProvider.asyncSaveLinks(processedLinks);
+    this._previewProvider.asyncSaveLinks(processedLinks, event);
```

Comment:
unfortunately in the tests, since we only pass one parameter to ```asyncSaveLinks```, it expects it to be the links, so we would have to pass an empty first parameter in the tests like ```asyncSaveLinks({}, links)```, which is kind of ugly